webapp/channels/src/components/channel_invite_modal/group_option/group_option.test.tsx

```diff
@@ -0,0 +1,122 @@
+
+import GroupOption from './group_option';
+
+const mockGroup = {
+    id: 'group-id',
+    display_name: 'Group Name',
+    name: 'groupname',
+    member_ids: ['user1', 'user2'],
+    member_count: 2,
+} as Group&Value;
+
+describe('GroupOption', () => {
+    let props: ComponentProps<typeof GroupOption>;
+
+    beforeEach(() => {
+        props = {
+            group: mockGroup,
+            isSelected: false,
+            rowSelected: '',
+            selectedItemRef: React.createRef(),
+            onMouseMove: jest.fn(),
+            addUserProfile: jest.fn(),
+        };
+    });
+
+    test('should match snapshot', () => {
+        const wrapper = renderWithContext(
+            <IntlProvider locale='en'>
+                <Provider store={store}>
+                    <GroupOption
+                        {...props}
+                    />
+                </Provider>
+            </IntlProvider>,
+
+        );
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it('should render correctly', () => {
+        const wrapper = renderWithContext(
+            <IntlProvider locale='en'>
+                <Provider store={store}>
+                    <GroupOption
+                        {...props}
+                    />
+                </Provider>
+            </IntlProvider>,
+
+        );
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it('should cleanup keydown event listener on unmount', () => {
+        const addUserProfileMock = jest.fn();
+        const wrapper = renderWithContext(
+            <IntlProvider locale='en'>
+                <Provider store={store}>
+                    <GroupOption
+                        group={mockGroup}
+                        isSelected={false}
+                        rowSelected=''
+                        selectedItemRef={React.createRef()}
+                        onMouseMove={() => {}}
+                        addUserProfile={addUserProfileMock}
+                    />
+                </Provider>
+            </IntlProvider>,
+
+        );
+        wrapper.unmount();
+        const event = new KeyboardEvent('keydown', {key: 'Enter'});
+        document.dispatchEvent(event);
+        expect(addUserProfileMock).not.toHaveBeenCalled();
+    });
+
+    it('should cleanup keydown event listener on unmount and dispatch event correctly before unmount', () => {
+        const addUserProfileMock = jest.fn();
+        props.isSelected = true;
+
+        const wrapper = renderWithContext(
+            <IntlProvider locale='en'>
+                <Provider store={store}>
+                    <GroupOption
+                        {...props}
+                        addUserProfile={addUserProfileMock}
+                    />
+                </Provider>
+            </IntlProvider>,
+        );
+
+        const event = new KeyboardEvent('keydown', {key: 'Enter'});
+        document.dispatchEvent(event);
+
+        setTimeout(() => {
```

Comment:
can we use split this into two `await waitFor` here instead? First one to check `addUserProfileMock` is called and after unmount & dispatch another wait to check `addUserProfileMock` is not called